### ISSUE TITLE
Explain release

### ISSUE THREAD

**Issue by chadwhitacre** [opencompany/www.opencompany.org#176]
@waldyrious Can I ask what's up with https://github.com/opencompany/www.opencompany.org/releases/tag/0 ? The `*.exe` smells like a potential security compromise with your GitHub account. 😞

**Comment by axifive** [comment]
@chadwhitacre Yes, it is. The Gitea had the same release:
https://github.com/go-gitea/gitea/issues/4167

**Comment by balupton** [comment]
While I'm not a maintainer, it seems I've got write access, so I've taken down the release. Here is a screenshot for posterity. Will reach out to @waldyrious now.

![screen shot 2018-06-08 at 7 51 36 pm](https://user-images.githubusercontent.com/61148/41156704-9eea530a-6b55-11e8-9d7c-a5a0c4c2007e.png)

**Comment by balupton** [comment]
I will be in https://appear.in/release-0-compromised while I attend to this

**Comment by balupton** [comment]
Ok, I am a member of the org, but not an owner. @waldyrious is on the owner team, as are 3 others. Pinging @opencompany/owners

**Comment by balupton** [comment]
Ok, I've sent an alert to github for the meantime:

![screen shot 2018-06-08 at 7 57 12 pm](https://user-images.githubusercontent.com/61148/41156840-2cca0a26-6b56-11e8-8f5a-b3975714508a.png)

Not sure what else I can do.

**Comment by balupton** [comment]
Immediately, until Github does something I guess having one of the other owners remove @waldyrious permissions for the meantime is the right step forward.

**Comment by balupton** [comment]
Looking through other orgs and repos he has access to, install.exe also appears other places:

- https://github.com/mwclient/mwclient/releases by @danmichaelo
- https://github.com/opencompany/opencompany.github.io/releases
- https://github.com/tldr-pages/tldr/releases by @rprieto
- https://github.com/tldr-pages/tldr-node-client/releases by @agnivade 
- https://github.com/tldr-pages/tldr-cpp-client/releases by @Leandros
- https://github.com/browserpad/browserpad/releases
- https://github.com/JuliaLangPt/tutorial_PT_BR/releases
- https://github.com/JuliaLangPt/julia/releases
- https://github.com/Julia-i18n/Gettext.jl/releases
- https://github.com/Julia-i18n/julialang.github.com/releases
- https://github.com/Julia-i18n/julia-i18n/releases
- https://github.com/wikimedia-portugal/wiki-migration-toolkit/releases
- https://github.com/wikimedia-portugal/dez-anos/releases

**Comment by balupton** [comment]
Pinging:

- @danmichaelo from mwclient
- @agnivade and @sbrl from tldr-pages
- @El-Zurdo-Sordo @felipenoris @SalchiPapa from JuliaLangPt

The other orgs are only waldy

**Comment by Leandros** [comment]
Thanks for the ping. I've removed the `install.exe' from my release (but not after downloading it, let's get IDA Pro warmed up and see what it is :P ).

**Comment by felipenoris** [comment]
Thank you! I've removed the file from JuliaLangPt org.

**Comment by balupton** [comment]
Okay. I've submitted a report via https://bounty.github.com that has gone on their hackerone:
https://hackerone.com/bugs?report_id=363401

That's all I can do.

**Comment by rprieto** [comment]
Thanks for the ping, I've also deleted the release with the EXE file from https://github.com/tldr-pages/tldr/releases.

**Comment by Leandros** [comment]
It's ... a bitcoin miner.

**Comment by agnivade** [comment]
What the hell just happened ? Is @waldyrious' account compromised ? Deleted the exe from node-client release.

@Leandros - Aha !

**Comment by balupton** [comment]
Alright. I'm signing off as it is evening here. If anyone still needs me, `balupton` on wire.com

Good work everyone

**Comment by justinclift** [comment]
No worries at all @balupton. :smile:

As a data point, @graystevens is the one who noticed the problem in your repo as mentioned here along with his initial analysis of the exe: https://github.com/go-gitea/gitea/issues/4167#issuecomment-395718026

**Comment by justinclift** [comment]
@Leandros That comment in the Gitea repo might be of interest to you too, as @graystevens has already done some initial analysis.

**Comment by graystevens** [comment]
Cheers for the nudge @justinclift. 

Some awesome information in here, seems we have some other repos compromised with the same binary - nice find.

@Leandros if you get anywhere with IDA let me know, be interested to know how close I got with my dynamic analysis.

**Comment by justinclift** [comment]
> Okay. I've submitted a report to GitHub via https://bounty.github.com that has gone on their hackerone: https://hackerone.com/bugs?report_id=363401

Just tried accessing that HackerOne URL.  It requires sign-up first, which seems a bit weird.

Oh well, I guess it's to stop the merely curious from looking. :smile:

**Comment by chadwhitacre** [comment]
Awesome work @balupton et al.! 👏

Closing since the release is gone from here, hopefully @waldyrious and GitHub get their situation sorted out soon.

**Comment by justinclift** [comment]
@waldyrious account page here on GitHub is 404-ing now:

&nbsp; &nbsp; https://github.com/waldyrious

But several of the repos - including this OpenCompany one - still have the malware showing. :frowning_face:

**Comment by justinclift** [comment]
@chadwhitacre Do you have edit/change access to this which still has the malware?

&nbsp; &nbsp; https://github.com/opencompany/opencompany.github.io/releases <-- still has malware

Note that the **tag** page version doesn't have the malware, while the **release** page version does:

&nbsp; &nbsp; https://github.com/opencompany/www.opencompany.org/releases/tag/0 <-- no malware

**Comment by justinclift** [comment]
Oh @balupton, you have write access so should be able to fix.  Sorry for keeping hassling you. :innocent:

**Comment by balupton** [comment]
No write access for that repo unfortunately. So someone else will need to do the removal.

Got a response from GitHub about 24 hours ago, that they were looking into the reports. So that would explain the 404’ing

**Comment by chadwhitacre** [comment]
Good catch @justinclift, reopening until https://github.com/opencompany/opencompany.github.io/releases is cleaned up.

Ping @galuszkak @tenkabuto @timothyfcook!

**Comment by galuszkak** [comment]
@chadwhitacre deleted.

Closing.

**Comment by chadwhitacre** [comment]
!m @galuszkak :o)

**Comment by sbrl** [comment]
Yeah,  @waldrious' account was compromised. They've now changed their password, enabled 2-factor authentication, and revoked all sessions, and has commented here, but until GitHub unblock their account said comment won't be visible.

Source: [tldr-pages Gitter channel](https://gitter.im/tldr-pages/tldr)

**Comment by waldyrious** [comment]
Hi all. My account has just been unflagged. GitHub support has confirmed that it is now secured with the measures I took. I am now awaiting a response regarding what could have caused this problem.

In any case, I apologize to everyone involved for the inconvenience. Many thanks for taking quick action!

In light of this incident, I wonder if more people should be granted owner access to the opencompany org (@balupton in particular). Let me know what you think.

ps - I edited @sbrl's comment above to fix a typo in my username :)

**Comment by justinclift** [comment]
@waldyrious Good that you're back. :smile:

Do you have access to change this release?

* https://github.com/mwclient/mwclient/releases/tag/v0.9.0

It's still showing the malware available for download. :frowning_face:

**Comment by waldyrious** [comment]
Update: I've deleted all the remaining releases mentioned in [this comment](https://github.com/opencompany/www.opencompany.org/issues/176#issuecomment-395740868), as well as:
- https://github.com/opencompany/awesome-open-company/releases
- https://github.com/waldyrious/wikispeed/releases
- https://github.com/zeitgeist-portugal/zeitgeist.pt/releases
- https://github.com/TeamOfOne/guitarjs/releases
- https://github.com/TeamOfOne/teamofone.github.io/releases
- https://github.com/improvebraga/ImproveBraga/releases
- https://github.com/improvebraga/venues/releases
- https://github.com/improvebraga/oqfb/releases
- https://github.com/improvebraga/laptop_coffee/releases

I believe that completes the full list of releases made with my account, so everything should be sorted out now. Thanks again everyone, for bearing with me.

----

Btw, I also deleted the tags using the CLI, since GitHub doesn't provide a way to do it in the web interface. For future reference, I created a two-liner script called `del-tag-0.sh` to streamline the process:

```sh
#!/bin/sh

git clone "git@github.com:$1.git" &&
(cd "$(echo "$1" | cut -d'/' -f2)" && git tag -d 0 && git push --delete origin 0)
```

which I then ran multiple times like this:

    . del-tag-0.sh opencompany/awesome-open-company

**Comment by waldyrious** [comment]
> Do you have access to change this release?
>
> * https://github.com/mwclient/mwclient/releases/tag/v0.9.0
>
> It's still showing the malware available for download. :frowning_face: 

Huh... that's odd. The release was created by @danmichaelo. I suppose my account was used to append the executable as an attachment to it? In any case, I edited the release and removed the malware from it.

**Comment by justinclift** [comment]
Thanks @waldyrious. :smile:

**Comment by waldyrious** [comment]
Thank you for the heads up :) I had missed that one for sure.

**Comment by balupton** [comment]
Glad you got your account back! that everything is becoming sorted, and that I could be of use 😊

> In light of this incident, I wonder if more people should be granted owner access to the opencompany org (@balupton in particular). Let me know what you think.

Whatever you decide I'm happy with 👍

**Comment by waldyrious** [comment]
Cool @balupton, I made you an owner. Cheers!

**Comment by justinclift** [comment]
> As the cause of the compromise seems to be a malicious script.

Hmmm, wouldn't it be more that some malicious script is taking advantage of bad opsec?

2FA is kind of a way to hedge that bet, but does so at the expense of usability.  The old security vs usability-for-users tradeoff. :smile:

Note - Not saying it's not warranted, I'm just pointing out that org-wide rollouts will change where the needle is on the usability scale so might exclude or reduce the involvement of some members.  Since OpenCompany isn't a huge org though, it's probably do-able. :smile:

**Comment by galuszkak** [comment]
I'm using Trezor for 2FA. You can always restore key from seed if you lost Your hardware key.

**Comment by sbrl** [comment]
Same with Authy. They have a security check or something you can use to regain access to your account. Don't know much more than that, as I've never had to use it :P
